internal/hcl/block.go

```diff
@@ -1082,3 +1090,116 @@ func loadBlocksFromFile(file file, schema *hcl.BodySchema) (hcl.Blocks, error) {
 func stripCount(s string) string {
 	return foreachRegex.ReplaceAllString(countRegex.ReplaceAllString(s, ""), "")
 }
+
+// BlockValueFunc defines a type that returns a set of fake/mocked values for a given block type.
+type BlockValueFunc = func(b *Block) cty.Value
+
+var (
+	defaultAWSRegion = "us-east-1"
+	defaultGCPRegion = "us-central1"
+
+	blockValueFuncs = map[string]BlockValueFunc{
+		"data.aws_availability_zones": awsAvailabilityZonesValues,
+		"data.google_compute_zones":   googleComputeZonesValues,
+		"resource.random_shuffle":     randomShuffleValues,
+	}
+)
+
+func randomShuffleValues(b *Block) cty.Value {
```

Comment:
It's not clear to me what we're doing here. From what I can tell we're essentially slicing the list - I assume that's so the result of this is always consistent?

Reply:
it's a function to mock the values returned from `resource.random_shuffle` https://github.com/hashicorp/terraform-provider-random/blob/main/docs/resources/shuffle.md. This resource uses the `result_count` attribute to return a slice of the `input` attribute, which is randomly sorted. We don't randomly sort it here as it's unecessary for us, and a consistent output is probably better for debuging/tests. I'll add a comment to clarify.